### PR TITLE
🎨 672 app button tiny connectivity

### LINKED ISSUE #672: Button - manage state according the connectivity
### Is your feature request related to a problem?

Add a param ``disableAccordingConnectivity`` (``false`` by default) in the widget "AppButtonTiny", 
Example:

The goal is to integrate directly in the widget the fact to disable it if there is no connectivity

```
if (accountSelected!.balance!.isNativeTokenValuePositive() &&
                    connectivityStatusProvider ==
                        ConnectivityStatus.isConnected)
                  AppButtonTiny(
                    AppButtonTinyType.primary,
                    localizations.createFungibleToken,
                    Dimens.buttonBottomDimens,
                    icon: Icon(
                      Icons.add,
                      color: theme.mainButtonLabel,
                      size: 14,
                    ),
                    key: const Key('createTokenFungible'),
                    onPressed: () {
                      Sheets.showAppHeightNineSheet(
                        context: context,
                        ref: ref,
                        widget: const AddTokenSheet(),
                      );
                    },
                  )
                else
                  AppButtonTiny(
                    AppButtonTinyType.primaryOutline,
                    localizations.createFungibleToken,
                    Dimens.buttonBottomDimens,
                    icon: Icon(
                      Icons.add,
                      color: theme.mainButtonLabel!.withOpacity(0.3),
                      size: 14,
                    ),
                    key: const Key('createTokenFungible'),
                    onPressed: () {},
                  ),
```

we should have: 

```
 AppButtonTiny(
                    AppButtonTinyType.primary,
                    localizations.createFungibleToken,
                    Dimens.buttonBottomDimens,
                    icon: Icon(
                      Icons.add,
                      color: theme.mainButtonLabel,
                      size: 14,
                    ),
                    key: const Key('createTokenFungible'),
                    disableAccordingConnectivity = true,
                    onPressed: () {
                      Sheets.showAppHeightNineSheet(
                        context: context,
                        ref: ref,
                        widget: const AddTokenSheet(),
                      );
                    },
                  )

```

### Describe the solution you'd like

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Esquimor**
# Description

Update AppButtonTiny to have directly the connectivity on it

Fixes #672 (issue)

## Type of change

Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality)
- Refactorization

## Material used:

Please delete options that are not relevant.
- Android (Smartphone/Tablet)

## How Has This Been Tested?

- Manually

## Checklist:

- My code follows the style guidelines of this project
- I have performed a self-review of my own code
- I have commented my code, particularly in hard-to-understand areas
- I have made corresponding changes to the documentation
- My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- New and existing unit tests pass locally with my changes
- Any dependent changes have been merged and published in downstream modules

## Useful info for the reviewer:

I put this in draft to have a first review before change all the code.

### CODE REVIEW

**Comment by Esquimor** [comment]
@redDwarf03 or @Chralu can you do a first review to see if my code is correct.

If yes, i will change all the button of the app.

If you don't have the time to review it, please tell me.

File: lib/ui/widgets/components/app_button_tiny.dart [hunk 1]
```diff
@@ -12,6 +13,57 @@ import 'package:flutter_vibrate/flutter_vibrate.dart';
 
 enum AppButtonTinyType { primary, primaryOutline }
 
+class AppButtonTinyConnectivity extends ConsumerWidget {
+  const AppButtonTinyConnectivity(
+    this.buttonText,
```
Comment (Chralu) [inline lib/ui/widgets/components/app_button_tiny.dart:18]: Maybe you should add a `AppButtonTinyType type` property. That property would be simply passed to `AppButtonTiny` constructor.

File: lib/ui/widgets/components/app_button_tiny.dart [hunk 2]
```diff
@@ -12,6 +13,57 @@ import 'package:flutter_vibrate/flutter_vibrate.dart';
...
+        icon: Icon(
+          icon,
+          color: (!disabled && isConnected)
+              ? theme.mainButtonLabel
+              : theme.mainButtonLabel!.withOpacity(0.3),
+          size: 14,
+        ),
+        key: key,
+        onPressed: onPressed,
+        disabled: disabled);
```
Comment (Chralu) [inline lib/ui/widgets/components/app_button_tiny.dart]: Linter asks for a trailing comma on this line.

**Review by Chralu — COMMENTED** [review]
Besides the two small comments, everything seems fine to me 👍

**Pull request closed without merging**